src/test/scala/cognite/spark/v1/AssetHierarchyBuilderTest.scala

```diff
@@ -1048,6 +1049,26 @@ class AssetHierarchyBuilderTest
     cleanDB(key)
   }
 
+  it should "successfully do batching of subtree roots" in {
+    val key = shortRandomString()
+
+    val dad = AssetCreate("dad", None, None, None, Some("dad"), None, Some(""))
+
+    val kids = (1 to 1100).flatMap(k => Seq(
+      AssetCreate(s"kid$k", None, None, None, Some(s"kid$k"), None, Some("dad"))))
+    val grandkids = (1 to 1100).flatMap(k => Seq(
+      AssetCreate(s"grandkid$k", None, None, None, Some(s"grandkid$k"), None, Some(s"kid$k"))
+    ))
```

Comment:
Small nit, but you can simplify `x.flatMap(x => Seq(...))` to `x.map(x => ...)`

Reply:
Good catch! I initially coded it up with several things in the Seq(), hence the weird logic here.